ncs/basic_kms.py

```diff
@@ -72,6 +82,73 @@ def encrypt(self, plaintext, key_name, context, aad) -> tuple[bytes, bytes, byte
 
         return nonce, tag, ciphertext
 
+    def _verify_signing_key_type(self, private_key, algorithm) -> bool:
+        """Verify if the key type matches the provided key."""
+        if isinstance(private_key, EllipticCurvePrivateKey):
+            return f"es-{private_key.key_size}" == algorithm
+        elif isinstance(private_key, Ed25519PrivateKey) or isinstance(private_key, Ed448PrivateKey):
+            return "eddsa" == algorithm
+        else:
+            raise ValueError(f"Key {type(private_key)} not supported")
+
+    def _create_cose_es_signature(self, input_data, private_key: bytes) -> bytes:
+        """Create ECDSA signature and return signature bytes."""
+        hash_map = {256: hashes.SHA256(), 384: hashes.SHA384(), 521: hashes.SHA512()}
+        dss_signature = private_key.sign(input_data, ec.ECDSA(hash_map[private_key.key_size]))
+        r, s = decode_dss_signature(dss_signature)
+        return r.to_bytes(math.ceil(private_key.key_size / 8), byteorder="big") + s.to_bytes(
+            math.ceil(private_key.key_size / 8), byteorder="big"
+        )
+
+    def _create_cose_ed_signature(self, input_data, private_key: bytes) -> bytes:
+        """Create ECDSA signature and return signature bytes."""
+        return private_key.sign(input_data)
+
+    def _get_sign_method(self, private_key) -> bool:
+        """Return sign method based on key type."""
+        if isinstance(private_key, EllipticCurvePrivateKey):
+            return self._create_cose_es_signature
+        elif isinstance(private_key, Ed25519PrivateKey) or isinstance(private_key, Ed448PrivateKey):
+            return self._create_cose_ed_signature
+        else:
+            raise ValueError(f"Key {type(private_key)} not supported")
+
+    def sign(self, data, key_name, algorithm, context) -> bytes:
+        """
+        Sign the data with a private key.
+
+        :param data: The data to be signed.
+        :param key_name: The name of the private key to be used.
+        :param algorithm: The name of the algorithm to be used.
+                          Used to verify if the key in the provided file contains a key of a compatible type.
+        :param context: The context to be used
+
+        :return: The signature.
+        :rtype: bytes
+        """
+        # TODO: support DER format
```

Comment:
Please either remove `TODO` or add a reference to the JIRA ticket.

Reply:
Added a couple of lines which should support both formats